storage/oasis/nodeapi/enigma/convert.go

```diff
@@ -0,0 +1,480 @@
+		Time:    d.Time,
+		ChainID: d.ChainID,
+		Governance: governance.Genesis{
+			Proposals:   proposals,
+			VoteEntries: voteEntries,
+		},
+		Registry: registry.Genesis{
+			Entities:          d.Registry.Entities,
+			Runtimes:          runtimes,
+			SuspendedRuntimes: suspendedRuntimes,
+			Nodes:             nodes,
+		},
+		Staking: staking.Genesis{
+			CommonPool:           d.Staking.CommonPool,
+			LastBlockFees:        d.Staking.LastBlockFees,
+			GovernanceDeposits:   d.Staking.GovernanceDeposits,
+			Ledger:               ledger,
+			Delegations:          delegations,
+			DebondingDelegations: debondingDelegations,
+		},
+	}, nil
+}
+
+func convertEvent(e txResultsEnigma.Event) nodeapi.Event {
+	ret := nodeapi.Event{}
+	switch {
+	case e.Staking != nil:
+		switch {
+		case e.Staking.Transfer != nil:
+			ret = nodeapi.Event{
+				StakingTransfer: (*nodeapi.TransferEvent)(e.Staking.Transfer),
+				RawBody:         common.TryAsJSON(e.Staking.Transfer),
+				Type:            apiTypes.ConsensusEventTypeStakingTransfer,
+			}
+		case e.Staking.Burn != nil:
+			ret = nodeapi.Event{
+				StakingBurn: (*nodeapi.BurnEvent)(e.Staking.Burn),
+				RawBody:     common.TryAsJSON(e.Staking.Burn),
+				Type:        apiTypes.ConsensusEventTypeStakingBurn,
+			}
+		case e.Staking.Escrow != nil:
+			switch {
+			case e.Staking.Escrow.Add != nil:
+				ret = nodeapi.Event{
+					StakingAddEscrow: (*nodeapi.AddEscrowEvent)(e.Staking.Escrow.Add),
+					RawBody:          common.TryAsJSON(e.Staking.Escrow.Add),
+					Type:             apiTypes.ConsensusEventTypeStakingEscrowAdd,
+				}
+			case e.Staking.Escrow.Take != nil:
+				ret = nodeapi.Event{
+					StakingTakeEscrow: &nodeapi.TakeEscrowEvent{
+						Owner:  e.Staking.Escrow.Take.Owner,
+						Amount: e.Staking.Escrow.Take.Amount,
+						// TODO: Expand signature of TakeEscrowEvent so we can store .DebondingAmount (assuming we care about it)
+					},
+					RawBody: common.TryAsJSON(e.Staking.Escrow.Take),
+					Type:    apiTypes.ConsensusEventTypeStakingEscrowTake,
+				}
+			case e.Staking.Escrow.Reclaim != nil:
+				ret = nodeapi.Event{
+					StakingReclaimEscrow: (*nodeapi.ReclaimEscrowEvent)(e.Staking.Escrow.Reclaim),
+					RawBody:              common.TryAsJSON(e.Staking.Escrow.Reclaim),
+					Type:                 apiTypes.ConsensusEventTypeStakingEscrowReclaim,
+				}
+			case e.Staking.Escrow.DebondingStart != nil: // Note: Event started appearing mid-Enigma.
+				ret = nodeapi.Event{
+					StakingDebondingStart: &nodeapi.DebondingStartEscrowEvent{
+						Owner:           e.Staking.Escrow.DebondingStart.Owner,
+						Escrow:          e.Staking.Escrow.DebondingStart.Escrow,
+						Amount:          e.Staking.Escrow.DebondingStart.Amount,
+						ActiveShares:    e.Staking.Escrow.DebondingStart.ActiveShares,
+						DebondingShares: e.Staking.Escrow.DebondingStart.DebondingShares,
+						DebondEndTime:   0, // Not provided in Enigma; added in core v22.0, i.e. the testnet-only Damask precursor.
```

Comment:
woa is it really not?

Reply:
The comment is False. Thank you, fixed. It was a remnant of the Cobalt code (from which this file is derived), combined with a little overly-automatic find-replace (Cobalt -> Enigma). I went through all the comments in this file and found another field that we can populate in Enigma but couldn't populate in Damask.